src/command.ts

```diff
@@ -1,28 +1,48 @@
-import { workspace } from "vscode";
-import { Argument } from "./provider";
-import { ensureTerminalExists, selectTerminal } from "./terminal";
+import { workspace, window, Terminal } from "vscode";
+import { createNewTerminal, ensureTerminalExists, selectTerminal } from "./terminal";
+import { ConfigService } from './configService';
 
-export const runMakeCommand = () => async (argument: Argument) => {
-  if(workspace.workspaceFolders) {
-    sendTextsToTerminal([
+export const runMakeCommand = () => async (argument: string) => {
+  if (workspace.workspaceFolders) {
+    sendTextsToTerminal(argument, [
       `cd ${workspace.workspaceFolders[0].uri.fsPath}/`,
       `make ${argument}`
     ]);
   }
 };
 
-const sendTextToTerminal = async (text: string) => {
-  if (ensureTerminalExists()) {
-    const terminal = await selectTerminal();
-
-    if (terminal) {
-      terminal.sendText(text);
-    }
+const sendTextsToTerminal = async (argument: string, texts: string[]) => {
+  var terminal = await determineTerminalToUse(argument);
+  if (!terminal)
+  {
+    return;
   }
+  texts.forEach(text => sendTextToTerminal(terminal!, text));
 };
 
-const sendTextsToTerminal = async (texts: string[]) => {
-  for (let i = 0; i < texts.length; i++) {
-    sendTextToTerminal(texts[i]);
-  }
+const sendTextToTerminal = async (terminal: Terminal, text: string) => {
+  terminal?.sendText(text);
 };
+
```

Comment:
New method to encapsulate the logic around which terminal commands should be sent to as this got more complex with the new options.